### ISSUE TITLE
Sync gets stuck if using job hooks with multiple replicas on argocd-application-controller

### ISSUE THREAD

**Issue by snowping** [argoproj/argo-cd#4886]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Sync operation on ArgoCD gets stuck when using PostSync hooks with delete policy 'BeforeHookCreation'. It only gets stuck if we enable sharding on the Application Controller (multiple replicase) 

**To Reproduce**

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

1. Enable sharding on Application Controller e.g. 6 replicas
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: argocd-application-controller
spec:
  replicas: 6 # TODO Sharding is not working with hooks yet! Workaround: only 1 replica!
  template:
    spec:
      containers:
      - name: argocd-application-controller
        env:
        - name: ARGOCD_CONTROLLER_REPLICAS
          value: "6" # Enable sharding (always to be matched with replica!)
```

2. Add multiple "PostSync" jobs to an ArgoCD application with the following annotations:
```
apiVersion: batch/v1
kind: Job
metadata:
  name: {{ .name }}-vault-secrets-init-job
  namespace: cluster-bootstrap
  labels:
    helm.sh/chart: "{{$root.Chart.Name}}-{{$root.Chart.Version}}"
  annotations:
    argocd.argoproj.io/hook: PostSync
    argocd.argoproj.io/hook-delete-policy: BeforeHookCreation
spec:
...
```

3. Push "sync" button on UI 

When running with Sharding the UI gets stuck with status "waiting for completion of hook batch/Job...". It stucks forever until new sync with option "prune".

**Expected behavior**

Jobs are running and as soon as they are completed sync status turns green

**Version**

```shell
argocd: v1.7.9+f6dc8c3
  BuildDate: 2020-11-17T23:18:20Z
  GitCommit: f6dc8c389a00d08254f66af78d0cae1fdecf7484
  GitTreeState: clean
  GoVersion: go1.14.12
  Compiler: gc
  Platform: linux/amd64
```

**Comment by jannfis** [comment]
Hi @snowping - controller sharding is available from the upcoming v1.8 release, and is not yet supported at all.

Sorry for the confusion, as it is already available from the documentation. This confuses quite a few people.

Versioning of documentation is on our priority list, please bear with us.

I'm closing this bug. Feel free to reopen if you encounter the same issue with v1.8.

**Comment by snowping** [comment]
HI @jannfis 

Thanks for the quick response and info.
